l2geth/internal/ethapi/api.go

```diff
@@ -780,9 +781,12 @@ func (s *PublicBlockChainAPI) GetBlockRange(ctx context.Context, startNumber rpc
 	// For each block in range, get block and append to array.
 	for number := startNumber; number <= endNumber; number++ {
 		block, err := s.GetBlockByNumber(ctx, number, fullTx)
-		if block == nil || err != nil {
+		if err != nil {
 			return nil, err
 		}
+		if block == nil {
+			return nil, ethereum.NotFound
```

Comment:
Why not use l2geth's `NotFound`?

https://github.com/ethereum-optimism/optimism/blob/develop/l2geth/interfaces.go#L29

Reply:
It's easier for a client to error check using l2geth's `NotFound` rather than depending on geth.

Reply:
Actually I think this should be a more aggressive custom error. Check above already confirms that the end of the block range is available, so if something in the middle is not available that suggests a more concerning problem. IMO this is a "should not happen" type of error.

Reply:
You are right that this should not use the L1 `NotFound`, my bad - `goimports` pulled that in on save and I should have double checked it. This is definitely a should not happen kind of error, so i can define a custom error type to return